src/main/java/com/openlattice/authorization/Ace.java

```diff
@@ -21,21 +21,41 @@
 import com.openlattice.client.serialization.SerializationConstants;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.OffsetDateTime;
 import java.util.EnumSet;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
 
 public class Ace {
-    private final Principal           principal;
-    private final EnumSet<Permission> permissions;
-    private transient int h = 0;
+    private final     Principal           principal;
+    private final     EnumSet<Permission> permissions;
+    private final     OffsetDateTime      expirationDate;
+    private transient int                 h = 0;
```

Comment:
What is `h`? Please use a more descriptive name.

Reply:
Actually, this is me and it's used as a cache for computing the hashcode fof the object since they are passed around in sets a lot.